src/config.ts

```diff
@@ -56,3 +98,23 @@ export const getConfigForFunction = ({
 
   return mergeOptions.apply({ concatArrays: true, ignoreUndefined: true }, matches)
 }
+
+const getFromFile = async (func: FunctionWithoutConfig): Promise<FunctionConfig> => {
+  const filename = `${basename(func.mainFile, extname(func.mainFile))}.json`
+  const configFilePath = join(dirname(func.mainFile), filename)
+
+  try {
+    const data = await fs.readFile(configFilePath, 'utf8')
+    const configFile = JSON.parse(data) as FunctionConfigFile
+
+    if (configFile.version === 1) {
```

Comment:
Curious as to how we will use the version field. Is it for us to know what version of functions bundling a framework could be running or can it also be used by a framework to track which config set was used with which release?

Reply:
I always like to version these files, because it makes it easy to introduce breaking changes in their format if we ever need to. If we do introduce new versions, consumers would check this field first and treat the rest of the file accordingly.

I would say it's very unlike that we'll ever need to introduce a breaking change in this file, but the price of adding this field is very low so I figured it was worth doing. It's totally fine to remove it if you don't see a need for it, though.

Reply:
It makes sense, I think it's better to keep it and not need it than not have it but need it later.